sdk/python/lib/pulumi/automation/_stack.py

```diff
@@ -606,6 +639,85 @@ def destroy(
+        
+        :param message: Message to associate with the import operation.
+        :param resources: The resources to import.
+        :param nameTable: \
+            The name table maps language names to parent and provider URNs. \
+            These names are used in the generated definitions, \
+            and should match the corresponding declarations \
+            in the source program. This table is required if any parents or providers are \
+            specified by the resources to import.
+        :param protect: Whether to protect the imported resources so that they are not deleted
+        :param generate_code: Whether to generate code for the imported resources
+        :param converter: The converter plugin to use for the import operation
+        :param converter_args: Additional arguments to pass to the converter plugin
+        :param on_output: A function to process the stdout stream.
+        :param show_secrets: Include config secrets in the ImportResult summary.
+        """
+        args = ["import", "--yes", "--skip-preview"]
+        if message is not None:
+            args.extend(["--message", message])
+
+        tempDir = ""
+        try:
+            tempDir = tempfile.mkdtemp(prefix="pulumi-import-")
+            if resources is not None:
+                importFilePath = os.path.join(tempDir, "import.json")
+                with open(importFilePath, mode="w", encoding="utf-8") as importFile:
+                    contents = {"resources": resources, "nameTable": nameTable}
+                    json.dump(contents, importFile)
+                    args.extend(["--file", importFilePath])
+
+            if protect is not None:
+                value = "true" if protect else "false"
+                args.append(f"--protect={value}")
+
+            generated_code_path = os.path.join(tempDir, "generated_code.txt")
+            if generate_code is False:
+                args.append("--generate-code=false")
+            else:
+                args.append(f"--out={generated_code_path}")
+
+            if converter is not None:
+                args.extend(["--from", converter])
+                if converter_args is not None:
+                    args.append("--")
+                    args.extend(converter_args)
+
+            import_result = self._run_pulumi_cmd_sync(args, on_output)
+            summary = self.info(show_secrets and not self._remote)
+            generated_code = ""
+            if generate_code is not False:
+                with open(generated_code_path, mode="r", encoding="utf-8") as codeFile:
+                    generated_code = codeFile.read()
+
+            assert summary is not None
+            return ImportResult(
+                stdout=import_result.stdout,
+                stderr=import_result.stderr,
+                generated_code=generated_code,
+                summary=summary,
+            )
+
+        finally:
+            if tempDir != "":
+                shutil.rmtree(tempDir)
```

Comment:
I feel like you should be able to instead wrap this block with something like:

```python
with tempfile.TemporaryDirectory() as tmp_dir:
  ...
```

Reply:
Nice catch! I used this now ✅